Lesson18/main.js

```diff
@@ -0,0 +1,213 @@
+const slideList = document.getElementById("js-slid-list");
+const slideWrapper = document.getElementById("js-slide-wrapper");
+
+const createElementWithClassName = (type, className) => {
+  const element = document.createElement(type);
+  element.className = className;
+  return element;
+}
+
+const createLoading = () => {
+  const loading = createElementWithClassName("img", "loading");
+  loading.id = "js-loading";
+  loading.src = "./loading-circle.gif";
+  return loading;
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const createErrorMessage = (error) => {
+  const errorMessage = createElementWithClassName("p", "error-message");
+  errorMessage.textContent = error;
+  return errorMessage;
+}
+
+const fetchErrorHandling = async (response) => {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+const getJsonOrError = async (url) => {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+const getSlideImgData = async () => {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(getJsonOrError("https://myjson.dit.upm.es/api/bins/gzcn")), 3000);
+  })
+}
+
+const tryGetSlideImgData = async () => {
+  slideWrapper.appendChild(createLoading());
+  try {
+    const slideImgData = await getSlideImgData();
+    if (slideImgData.length === 0) {
+      throw new Error("No data...");
+    }
+    return slideImgData;
+  } catch (e) {
+    slideWrapper.appendChild(createErrorMessage(e));
+  } finally {
+    removeLoading();
+  }
+}
+
+const init = async () => {
+  const imgData = await tryGetSlideImgData();
+  if (imgData) {
+    initSlideItem(imgData);
+    initArrowButtons();
+    initIndicator(imgData);
+    slideWrapper.appendChild(createCounterOfSlide(imgData));
+  }
+}
+
+init();
+
+const initSlideItem = (imgSources) => {
+  for (let i = 0; i < imgSources.length; i++) {
+    const createdSlideItem = createSlideItem(imgSources[i]);
+    /**
+     * Set the is-displaying class to the first element by default 
+     */
+    i === 0 && createdSlideItem.classList.add("is-displaying");
+    slideList.appendChild(createdSlideItem);
+  }
+}
+
+const initArrowButtons = () => {
+  const createdArrowButtons = createArrowButtons();
+  slideWrapper.appendChild(createdArrowButtons);
+  setClickEventInArrowButton();
+}
+
+const initIndicator = (data) => {
+  slideWrapper.appendChild(createIndicator(data));
```

Comment:
imagesにして& lengthしか使っていなさそうなので
```suggestion
  slideWrapper.appendChild(createIndicator(images.length));
```

Reply:
Thanks for the review.
You're right...
I fixed it.
57959a2